### PR TITLE
Test for firewalld container in ALP product

### PR DESCRIPTION
**Pull request by ilmanzo**
Introduce a new multi-machine test for checking containerized firewalld workload in ALP product

- Fixes also a bug on `mm_network.pm` that sporadically caused `lo` interface being selected

- Related ticket: https://progress.opensuse.org/issues/131189
- Needles: none
- Verification runs:
  - https://openqa.suse.de/tests/12908711
  - https://openqa.suse.de/tests/12908712
  - https://openqa.suse.de/tests/12908713
  - https://openqa.suse.de/tests/12908714
  - https://openqa.suse.de/tests/12908715
  - https://openqa.suse.de/tests/12908716
  - https://openqa.suse.de/tests/12908717
  - https://openqa.suse.de/tests/12908718
  - https://openqa.suse.de/tests/12908719
  - https://openqa.suse.de/tests/12908720
  - https://openqa.suse.de/tests/12908721
  - https://openqa.suse.de/tests/12908722
  - https://openqa.suse.de/tests/12908723
  - https://openqa.suse.de/tests/12908724
  - https://openqa.suse.de/tests/12908725
  - https://openqa.suse.de/tests/12908726
  - https://openqa.suse.de/tests/12908727
  - https://openqa.suse.de/tests/12908728
  - https://openqa.suse.de/tests/12908729
  - https://openqa.suse.de/tests/12908730
  - https://openqa.suse.de/tests/12908731
  - https://openqa.suse.de/tests/12908732
  - https://openqa.suse.de/tests/12908733
  - https://openqa.suse.de/tests/12908734
  - https://openqa.suse.de/tests/12908735
  - https://openqa.suse.de/tests/12908736
  - https://openqa.suse.de/tests/12908737
  - https://openqa.suse.de/tests/12908738
  - https://openqa.suse.de/tests/12908739
  - https://openqa.suse.de/tests/12908740

### CODE REVIEW

**Review by ricardobranco777 — APPROVED** [review]
Nice work!

**Comment by clanig** [comment]
Nice PR, thank you.
Could you please outsource the exclusion of the loopback interface in the library in a separate commit? Other than that everything looks fine to me.
From my perspective the VRs all look the same.

File: tests/microos/workloads/firewalld-container/firewalld_client.pm [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+#
+# Package: podman firewalld-container
+# Summary: install and verify firewalld container.
+# Maintainer: QE Core <qe-core@suse.de>
+
+use base 'consoletest';
+use warnings;
+use strict;
+use testapi;
+use serial_terminal 'select_serial_terminal';
```
Comment (dzedro) [inline tests/microos/workloads/firewalld-container/firewalld_client.pm:14]: When serial terminal would use that would be better.

File: tests/microos/workloads/firewalld-container/firewalld_client.pm [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+use lockapi;
+use utils qw(set_hostname);
+use mm_network 'setup_static_mm_network';
+use utils 'script_retry';
```
Comment (dzedro) [inline tests/microos/workloads/firewalld-container/firewalld_client.pm:18]: Functions in one import are cleaner.

Comment (ilmanzo) [reply tests/microos/workloads/firewalld-container/firewalld_client.pm]: good point, thanks

File: tests/microos/workloads/firewalld-container/firewalld_server.pm [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    firewall_port('open');
+    barrier_wait 'FIREWALLD_SERVER_PORT_OPEN';
+    # tells the client server is ready so can probe the port
+    # client is checking the server port
+    firewall_port('close');
+    barrier_wait 'FIREWALLD_SERVER_PORT_CLOSED';
+    # here client is checking the port
+    barrier_wait 'FIREWALLD_TEST_FINISHED';
+    assert_script_run "kill $python_pid";
+    assert_script_run "podman kill firewalld";
```
[the remark this answers is not in the document]
Comment (ilmanzo) [reply tests/microos/workloads/firewalld-container/firewalld_server.pm]: Well the test is finished so machine will be turned off, but if you prefer I can change :)

File: tests/microos/workloads/firewalld-container/firewalld_client.pm (hunk 1 above)
Comment (ilmanzo) [reply tests/microos/workloads/firewalld-container/firewalld_client.pm]: well spotted, just a leftover during development... Thanks

**Review by mloviska — APPROVED** [review]

**Comment by mloviska** [comment]
It is a pity that the test can be used only in qemu and also not sure about other archs. Other than that, it looks well! thanks

**Review by dzedro — APPROVED** [review]

**Comment by foursixnine** [comment]
> It is a pity that the test can be used only in qemu and also not sure about other archs. Other than that, it looks well! thanks

doesn't sound like a bad idea to extend it :), but first lets unify the multiple network stacks, so they have similar test coverage of the features...

File: lib/mm_network.pm [hunk 4]
```diff
@@ -128,7 +128,7 @@ sub configure_static_dns {
     my $servers = join(" ", @{$conf->{nameserver}});
 
     if ($is_nm) {
-        $nm_id = script_output('nmcli -t -f NAME c | head -n 1') unless ($nm_id);
+        $nm_id = script_output('nmcli -t -f NAME c | grep -v ^lo: | head -n 1') unless ($nm_id);
```
Comment (foursixnine) [inline lib/mm_network.pm:131]: you could even do that on the perl side, but we can leave that for a follow up ticket, same for the ones earlier

**Review by foursixnine — APPROVED** [review]

**Comment by foursixnine** [comment]
@ilmanzo feel free to merge :D

File: tests/microos/workloads/firewalld-container/firewalld_server.pm (hunk 3 above)
[the remark this answers is not in the document]
Comment (ricardobranco777) [reply tests/microos/workloads/firewalld-container/firewalld_server.pm]: The container was stopped but not removed afterwards. :(

**Pull request merged**